musig2test.c

```diff
@@ -40,114 +26,204 @@ int main(void) {
     printf("* Number of messages\t: %d \n", NR_MSGS);
     printf("--------------------------------------------------------------------------- \n");
 
+    /* musig2test parameters. */
+    int i,j,k,l;
+    int ind;
+    secp256k1_pubkey pk_list[N];    // Signers' public key list
+    secp256k1_pubkey batch_list[N*V*NR_MSGS];   // Stores the batches of signers
+    unsigned char *parsig_list[N];  // The list of partial signatures
 
 
-    /* Create a keypair for each signer ******************************/
+    musig2_context_sig *mcs_list[N]; // Array that holds N musig2_context_sig
+
+    /**** Initialization ****/
     printf("\n-------- Initialize Signers ----------------------------------------------- \n");
-    for(i=0; i<N; i++){
-        signer_LIST[i] = malloc(sizeof (SIGNER));
-        signer_LIST[i]->keypair = malloc(sizeof (secp256k1_keypair));
-        MuSig2_KeyGen(ctx, signer_LIST[i]->keypair);
+    for (i=0; i<N; i++){
+        mcs_list[i] = malloc(sizeof (musig2_context_sig));
+        mcs_list[i]->mc = malloc(sizeof (musig2_context));
+        mcs_list[i]->mc->ctx = secp256k1_context_clone(ctx);
```

Comment:
In this case we need to pass `ctx` as a parameter to `init` function. We can of course do that but I couldn't be sure to pass that as parameter, what do you think?

Reply:
I think that is better. Otherwise, the users of the library will have to allocate memory, which does not seem very usable. As a user of a crypto library (or a library in general) you don't want to be allocating memory unless it is necessary. I don't think that requiring memory allocation is worth it here.